validation_components/validation.py

```diff
@@ -6,23 +6,64 @@ def validation_runner(arguments: argparse.Namespace):
     loader = SpreadsheetLoader(arguments.spreadsheet)
     all_entries = loader.load()
 
+    error_list = verify_entries(all_entries)
+
+    if len(error_list) > 0:
+        print('Errors found within manifest:\n\t' + '\n\t'.join(error_list) + '\nPlease correct mistakes and validate again.')
+    else:
+        print('Manifest successfully validated, no errors found!')
+
+
+def verify_entries(all_entries):
     error_list = []
-    registered_values = set()
+    registered_values = {'__null____null__': [1, None, None]}
     timestamp = None
     for manifest_entry in all_entries:
-        if not manifest_entry.common_name or not manifest_entry.taxon_id:
-            error_code = 1
-            error_list.append(manifest_entry.report_error(error_code))
-        elif manifest_entry.query_id in registered_values:
-            pass
+        if manifest_entry.query_id in registered_values.keys():
+            error_code = registered_values[manifest_entry.query_id][0]
+            common_name_statement = registered_values[manifest_entry.query_id][1]
+            taxon_id_statement = registered_values[manifest_entry.query_id][2]
+            if error_code != 0:
+                error_list.append(manifest_entry.report_error(error_code, common_name_statement, taxon_id_statement))
+
         else:
             connecter = NcbiQuery()
-            timestamp = connecter.generate_new_timestamp(timestamp)
-            error_code, ncbi_common_name = connecter.query_ncbi(manifest_entry)
-            if error_code:
-                error_list.append(manifest_entry.report_error(error_code, ncbi_common_name))
-            registered_values.add(manifest_entry.query_id)
-    if len(error_list) > 0:
-        print('Errors found within manifest:\n\t' + '\n\t'.join(error_list) + '\nPlease correct mistakes and validate again.')
+
+            common_name_statement, ncbi_taxon_id, timestamp = resolve_common_name(connecter, manifest_entry,
+                                                                                  timestamp)
+
+            if manifest_entry.taxon_id == ncbi_taxon_id:
```

Comment:
you're overwriting your common_name_statement in this branch, which suggests to me that it shouldn't have been computed yet at all